.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,273 @@
+name: Foundry Benchmarks
+
+on:
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: "PR number to comment on (optional)"
+        required: false
+        type: string
+      versions:
+        description: "Comma-separated list of Foundry versions to benchmark (e.g., stable,nightly,v1.0.0)"
+        required: false
+        type: string
+        default: "stable,nightly"
+      repos:
+        description: "Comma-separated list of repos to benchmark (e.g., ithacaxyz/account:main,Vectorized/solady)"
+        required: false
+        type: string
+        default: "ithacaxyz/account:v0.3.2,Vectorized/solady:v0.1.22"
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  forge-test:
+    name: Run forge_test and forge_fuzz_test benchmarks
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
```

Comment:
building foundry-bench should happen only once, you can run the individual benchmarks in parallel after they pull the artifact